### DISCUSSION TITLE
ngrinder 사용시 문제

### DISCUSSION THREAD

**Discussion by Jung-Woo-sik** [naver/ngrinder#939] [Q&A]
ngrinder로 json 메시지를 post 하는 테스트 진행을 하고 있는데 2가지 문제점이 존재합니다

ngrinder version : 3.5.7
os : ubuntu
java : openjdk version "11.0.16" 
- ngrinder 테스트를 시간으로 하였을때 테스트 결과에서 표시되는 총 실행 테스트 보다 실제 테스트는 더 실행이 많이 되있는 문제
- ngrinder 에서 post하지 않았는데 에러가 추가되어있고 로그에는 남겨져 있지 않은 문제
    (혹시 결과창에 나오는 로그 외에 다른 로그가 쌓이는지)

이 문제 때문에 테스트 신뢰성에 문제가 생기는데 해결 방법이 있는지요

**Comment by donggyu04** [comment]
@Jung-Woo-sik 

1. ngrinder 테스트를 시간으로 하였을때 테스트 결과에서 표시되는 총 실행 테스트 보다 실제 테스트는 더 실행이 많이 되있는 문제
  - 100% 정확할 순 없습니다. 지표 수집 시 약간의 오차가 발생할 순 있는데요. ngrinder 사용하는 목적이 서버 부하테스트 이기 때문에 약간의 지표 오차는 부하 테스트에 큰 문제가 되지 않을것으로 생각됩니다. 

2. post 하지 않았는데 에러가 추가되어 있다는건 좀 이해가 안됩니다. 해당 에러가 ngrinder 에서 보낸 요청이 맞나요? post 하지 않았는데 에러가 어떻게 추가됐는지 의문입니다. 그리고 에이전트 로그는 1MB 기준으로 rotate 됩니다. 따라서 오류 로그가 rotate 된다면 에이전트 로그에 안보일 수 있는데요.
https://github.com/naver/ngrinder/wiki/How-to-adjust-logger-level 참고하셔서 로그 레벨을 수정하시면 될 것 같습니다.

3. ngrinder 는 현재 사내에서 다양한 사내 시스템을 타겟으로 40만건의 부하테스트를 진행하였고 현재도 사용되고 있습니다. 부하 테스트 신뢰성에 대해서는 크게 의심 안하셔도 될 것 같습니다.
